From 69e440f9b7a0e9a43ef582d4bb521722b448a7c2 Mon Sep 17 00:00:00 2001
From: Maxime Ripard <maxime.ripard@free-electrons.com>
Date: Mon, 31 Jul 2017 10:08:46 -0400
Subject: [PATCH] fix-getty-unit
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Prefer getty to agetty in console setup systemd units

Signed-off-by: Maxime Ripard <maxime.ripard@free-electrons.com>
Signed-off-by: Eric Le Bihan <eric.le.bihan.dev@free.fr>
[aduskett@gmail.com: Update for systemd v237]
Signed-off-by: Adam Duskett <aduskett@gmail.com>
[Jérémy: replace additional usage of agetty by getty.]
Signed-off-by: Jérémy Rosen <jeremy.rosen@smile.fr>
---
 units/console-getty.service.m4    | 2 +-
 units/container-getty@.service.m4 | 2 +-
 units/getty@.service.m4           | 5 +----
 units/serial-getty@.service.m4    | 2 +-
 4 files changed, 4 insertions(+), 7 deletions(-)

diff --git a/units/console-getty.service.m4 b/units/console-getty.service.m4
index 3c553240a..fd5ad9456 100644
--- a/units/console-getty.service.m4
+++ b/units/console-getty.service.m4
@@ -23,7 +23,7 @@ ConditionPathExists=/dev/console
 # The '-o' option value tells agetty to replace 'login' arguments with an
 # option to preserve environment (-p), followed by '--' for safety, and then
 # the entered username.
-ExecStart=-/sbin/agetty -o '-p -- \\u' --noclear --keep-baud console 115200,38400,9600 $TERM
+ExecStart=-/sbin/getty -L %I 115200 vt100
 Type=idle
 Restart=always
 UtmpIdentifier=cons
diff --git a/units/container-getty@.service.m4 b/units/container-getty@.service.m4
index 087ab7f9b..30f7b66fe 100644
--- a/units/container-getty@.service.m4
+++ b/units/container-getty@.service.m4
@@ -28,7 +28,7 @@ Before=rescue.service
 # The '-o' option value tells agetty to replace 'login' arguments with an
 # option to preserve environment (-p), followed by '--' for safety, and then
 # the entered username.
-ExecStart=-/sbin/agetty -o '-p -- \\u' --noclear --keep-baud pts/%I 115200,38400,9600 $TERM
+ExecStart=-/sbin/getty -L %I 115200 vt100
 Type=idle
 Restart=always
 RestartSec=0
diff --git a/units/getty@.service.m4 b/units/getty@.service.m4
index 80e793bb7..385758c61 100644
--- a/units/getty@.service.m4
+++ b/units/getty@.service.m4
@@ -35,10 +35,7 @@ ConditionPathExists=/dev/tty0
 
 [Service]
 # the VT is cleared by TTYVTDisallocate
-# The '-o' option value tells agetty to replace 'login' arguments with an
-# option to preserve environment (-p), followed by '--' for safety, and then
-# the entered username.
-ExecStart=-/sbin/agetty -o '-p -- \\u' --noclear %I $TERM
+ExecStart=-/sbin/getty -L %I 115200 vt100
 Type=idle
 Restart=always
 RestartSec=0
diff --git a/units/serial-getty@.service.m4 b/units/serial-getty@.service.m4
index 757b86ab2..3d60efdb6 100644
--- a/units/serial-getty@.service.m4
+++ b/units/serial-getty@.service.m4
@@ -33,7 +33,7 @@ Before=rescue.service
 # The '-o' option value tells agetty to replace 'login' arguments with an
 # option to preserve environment (-p), followed by '--' for safety, and then
 # the entered username.
-ExecStart=-/sbin/agetty -o '-p -- \\u' --keep-baud 115200,38400,9600 %I $TERM
+ExecStart=-/sbin/getty -L %I 115200 vt100
 Type=idle
 Restart=always
 UtmpIdentifier=%I
-- 
2.14.4

